simplecpp.cpp

```diff
@@ -3382,6 +3416,15 @@ void simplecpp::preprocess(simplecpp::TokenList &output, const simplecpp::TokenL
 
     std::set<std::string> pragmaOnce;
 
+    // Support an odd #pragma once usage to prevent unwanted inclusions from happening even once
+    // Example:
+    // #ifdef CPPCHECK
+    // #   pragma once "boost/*"
+    // #   pragma once "google/protobuf/*"
+    // #   pragma once "*.pb.h"
+    // #endif
+    std::set<Mask> pragmaOddOnce;
```

Comment:
well it could be `std::set<std::string> pragmaOnceMasks;` ..  instead of creating a Mask class what would be the downside to just provide a regular function like:

    bool match(const std::string& mask, const std::string& value, bool i);